src/conda_package_handling/tarball.py

```diff
@@ -59,6 +59,11 @@ def order(f):
         s2 = set(files_list)
         if len(s1) > len(s2):
             files_list.extend(s1 - s2)
+        # move info/ to front
+        files_list = [f for _, f in
+            sorted(enumerate(files_list),
+            key=lambda fi: (not fi[1].startswith('info' + os.path.sep), fi[0]))
+        ]
```

Comment:
I'm not following why this needs to use `enumerate` to sort `files_list` here, when you ignore it in the list comprehension.

Reply:
I'm sorting by whether it is info fi[1] then its enumeration fi[0] to try to preserve the old sort order.

Reply:
Does Python have a stable sort guarantee so I could only sort by whether the filename starts with 'info' and keep the relative ordering of everything else?

Reply:
I was thinking something like:

```python
from itertools import chain

is_info = lambda path: path.startswith("info" + os.path.sep)
files_list = list(
    chain(filter(is_info, files_list), filter(lambda x: not is_info(x), files_list))
)
```

But looking at it, it may not be as readable as I'd like.

Reply:
Yes the sort algorithm is [stable](https://wiki.python.org/moin/HowTo/Sorting/#Sort_Stability_and_Complex_Sorts). So this can be simplified.

Reply:
I feel like I should have known this 😬 Thanks Ken!